### PR TITLE
Slightly adjust docs about S3 incompatibilities

### PR DESCRIPTION
**Pull request by DaveCTurner**
It's often useful to quote these docs to users encountering problems
with their not-quite-S3-compatible storage system. In practice we don't
need to quote the bits in the middle but we do need the last sentence
about working with the supplier to address incompatibilities. This
commit reorders things so that the most commonly quoted sentences form a
standalone paragraph.

### CODE REVIEW

File: docs/reference/snapshot-restore/repository-s3.asciidoc [hunk 1]
```diff
@@ -226,16 +226,17 @@ emulate S3's behaviour in full. The `repository-s3` type requires full
...
 You can perform some basic checks of the suitability of your storage system
 using the {ref}/repo-analysis-api.html[repository analysis API]. If this API
 does not complete successfully, or indicates poor performance, then your
 storage system is not fully compatible with AWS S3 and therefore unsuitable for
-use as a snapshot repository. You will need to work with the supplier of your
-storage system to address any incompatibilities you encounter.
+use as a snapshot repository. However these checks do not guarantee full
+compatibility: incompatible error codes, consistency or performance may be
+particularly hard to track down since errors, consistency failures, and
+performance issues are often rare and hard to reproduce.
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/repository-s3.asciidoc:239]:
```suggestion
use as a snapshot repository. However, these checks do not guarantee full
compatibility. incompatible error codes, consistency or performance may be
hard to track down. These issues are typically rare and hard to reproduce.
```

**Review by jrodewig — APPROVED** [review]
LGTM. Left a minor suggestion you can take or leave as wanted. Thanks!

**Comment by DaveCTurner** [comment]
Thanks James, I took your shorter sentences and rejigged them a little more in d8957cd1e13caf5c9ba5f1cf881a8a6560987db3.

**Review by jrodewig — APPROVED** [review]
Changes look great! Thanks @DaveCTurner!

**Pull request merged**